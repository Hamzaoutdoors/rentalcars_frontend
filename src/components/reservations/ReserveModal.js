/* eslint-disable react/jsx-props-no-spreading */
import PropTypes from 'prop-types';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import IconButton from '@mui/material/IconButton';
import CloseIcon from '@mui/icons-material/Close';
import Typography from '@mui/material/Typography';
import { HowToRegOutlined } from '@mui/icons-material';
import { styled } from '@mui/material/styles';
import { useSelector } from 'react-redux';
import ReserveForm from './ReserveForm';
import store from '../../redux/configureStore';
import { expandModal } from '../../redux/reservations/reservationsSlice';
import { mobile } from '../../responsive';

const ReserveBtn = styled(Button)`
  padding: 0.5rem;
  border-radius: 20px;
  width: 150px;
  background-color: #97BF11 !important;
  text-decoration: none;
  text-align: center !important;
  color: white !important;
  cursor: pointer;

  &:hover {
    color: white !important;
    background-color: #9fcf01 !important;
  }
  ${mobile({
    margin: '0 1rem 2rem 0',
  })}
`;

const Title = styled(DialogTitle)`
  background-color: #f6a40e; 
  text-align: center;
  font-size: 1.5rem;
  font-weight: bold;
`;

const Content = styled(DialogContent)`
  padding: 0.5rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  & > .desc {
    font-size: 1rem;
    font-weight: bold;
    font-family: 'Urbanist', sans-serif;
  }`;

const DateContainer = styled(Typography)`
  display: flex;
  font-size: 16px;
  width: 100%;
`;

const BootstrapDialog = styled(Dialog)(({ theme }) => ({
  '& .MuiDialogContent-root': {
    padding: theme.spacing(3),
    boxShadow: '0px 0px 10px 0px rgba(0,0,0,0.75)',
  },
  '& .MuiDialogActions-root': {
    padding: theme.spacing(2),
    boxShadow: '0px 0px 10px 0px rgba(0,0,0,0.75)',
  },
}));

const BootstrapDialogTitle = (props) => {
  const { children, onClose, ...other } = props;

  return (
    <Title sx={{ m: 0, p: 2 }} {...other}>
      {children}
      {onClose ? (
        <IconButton
          aria-label="close"
          onClick={onClose}
          sx={{
            position: 'absolute',
            right: 8,
            top: 8,
            color: (theme) => theme.palette.grey[500],
          }}
        >
          <CloseIcon />
        </IconButton>
      ) : null}
    </Title>
  );
};

BootstrapDialogTitle.propTypes = {
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired,
};

const ReserveModal = () => {
  const { openModal } = useSelector((state) => state.reservations.utils);

  const toggleModal = () => {
    store.dispatch(expandModal(!openModal));
  };

  return (
    <>
      <ReserveBtn variant="outlined" onClick={toggleModal}>
        <HowToRegOutlined />
        Reserve
      </ReserveBtn>

      <BootstrapDialog
        onClose={toggleModal}
        aria-labelledby="customized-dialog-title"
        open={openModal}
      >
        <BootstrapDialogTitle id="customized-dialog-title" onClose={toggleModal}>
          Reserve Car
        </BootstrapDialogTitle>
        <Content dividers>
          <Typography gutterBottom className="desc">
            Reserve this car for a period of time.
          </Typography>
          <DateContainer variant="div">
            <ReserveForm />
          </DateContainer>
        </Content>
      </BootstrapDialog>
    </>
  );
};

export default ReserveModal;
