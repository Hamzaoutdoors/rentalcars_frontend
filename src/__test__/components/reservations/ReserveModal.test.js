import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import ReserveModal from '../../../components/reservations/ReserveModal';

it('ReserveModal component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <ReserveModal />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
