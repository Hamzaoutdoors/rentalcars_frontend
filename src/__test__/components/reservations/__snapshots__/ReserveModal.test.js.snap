// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReserveModal component renders correctly 1`] = `
<button
  className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1k063vf-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="HowToRegOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zM5 18c.2-.63 2.57-1.68 4.96-1.94l2.04-2c-.39-.04-.68-.06-1-.06-2.67 0-8 1.34-8 4v2h9l-2-2H5zm15.6-5.5-5.13 5.17-2.07-2.08L12 17l3.47 3.5L22 13.91z"
    />
  </svg>
  Reserve
</button>
`;
